### PR TITLE
executor: fix wrong peer count and improper handling of partitioned table in `TABLE_STORAGE_STATS` table

### PR DESCRIPTION
**Pull request by time-and-fate**
### What problem does this PR solve?

Issue Number: close #42611 close #42619

Problem Summary:

Please see the issues.

### What is changed and how it works?

1. To get the total peer count, add up all values of `PDRegionStats.StorePeerCount`, instead of using `len(PDRegionStats.StorePeerCount)`.
Because `PDRegionStats.StorePeerCount` is a map of `store id -> peer count`.
3. Get information and output a row for each partition.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

```sql
use test;
CREATE TABLE `t` (
  `a` int(11) DEFAULT NULL,
  `b` int(11) DEFAULT NULL,
  `c` int(11) DEFAULT NULL,
  KEY `ia` (`a`),
  KEY `ib` (`b`),
  KEY `ic` (`c`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
PARTITION BY RANGE (`a`)
(PARTITION `p0` VALUES LESS THAN (300),
 PARTITION `p1` VALUES LESS THAN (600),
 PARTITION `p2` VALUES LESS THAN (900),
 PARTITION `p3` VALUES LESS THAN (MAXVALUE));
```
insert some data into table t and split table.

Before:
```sql
> select * from information_schema.TABLE_STORAGE_STATS where table_schema='test' and table_name='t';
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
| TABLE_SCHEMA | TABLE_NAME | TABLE_ID | PEER_COUNT | REGION_COUNT | EMPTY_REGION_COUNT | TABLE_SIZE | TABLE_KEYS |
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
| test         | t          |       88 |          3 |            1 |                  1 |          1 |          0 |
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
```
After:
```sql
> select * from information_schema.TABLE_STORAGE_STATS where table_schema='test' and table_name='t';
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
| TABLE_SCHEMA | TABLE_NAME | TABLE_ID | PEER_COUNT | REGION_COUNT | EMPTY_REGION_COUNT | TABLE_SIZE | TABLE_KEYS |
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
| test         | t          |       88 |          3 |            1 |                  1 |          1 |          0 |
| test         | t          |       89 |          3 |            1 |                  0 |          9 |      74993 |
| test         | t          |       90 |         33 |           11 |                 10 |         17 |      47363 |
| test         | t          |       91 |          3 |            1 |                  0 |         17 |          0 |
| test         | t          |       92 |          3 |            1 |                  1 |          1 |          0 |
+--------------+------------+----------+------------+--------------+--------------------+------------+------------+
```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/infoschema_cluster_table_test.go [hunk 1]
```diff
@@ -326,3 +326,37 @@ func TestTableStorageStats(t *testing.T) {
...
+		s.store.(helper.Storage),
+		mockAddr,
+	}
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+
+	tk.MustExec("create table t (a int, b int, index idx(a))")
+	tk.MustQuery("select * from information_schema.TABLE_STORAGE_STATS where TABLE_SCHEMA = 'test' and TABLE_NAME='t'").Check(
+		testkit.Rows("test t 88 1 1 1 1 1"))
```
Comment (Defined2014) [inline executor/infoschema_cluster_table_test.go:343]: If you add another system table, the tableID will no longer be 88. This unit test often needs to be modified. Maybe not check tableID is better.

Comment (time-and-fate) [reply executor/infoschema_cluster_table_test.go]: I removed the table IDs. PTAL~

File: executor/infoschema_reader.go [hunk 2]
```diff
@@ -2093,25 +2093,37 @@ func (e *tableStorageStatsRetriever) setDataForTableStorageStats(ctx sessionctx.
...
+			}
+			peerCount := 0
+			for _, cnt := range e.stats.StorePeerCount {
+				peerCount += cnt
+			}
+
+			record := types.MakeDatums(
+				tbl.db,              // TABLE_SCHEMA
+				tbl.Name.O,          // TABLE_NAME
+				tableID,             // TABLE_ID
```
Comment (mjonss) [inline executor/infoschema_reader.go:2118]: Should we add a PARTITION_NAME as well for usability?

**Review by Defined2014 — APPROVED** [review]

File: executor/infoschema_reader.go (hunk 2 above)
Comment (mjonss) [reply executor/infoschema_reader.go:2118]: I agree, could you create a feature request for it?

**Review by mjonss — APPROVED** [review]
LGTM

**Review by mjonss — APPROVED** [review]

File: executor/infoschema_reader.go (hunk 2 above)
Comment (time-and-fate) [reply executor/infoschema_reader.go:2118]: Sure.
I created one issue: https://github.com/pingcap/tidb/issues/42860

**Pull request merged**